File: AngularNewsFrontEnd/src/app/article.ts
import { CommentDTO } from './comment-dto';
export interface Article {
  id: number;
  title: string;
  body: string;
  publishedAt: Date;
  views: number;
  rating: number;
  category: ArticleCategory;
  authorId: number;
  authorName: string;
  comments: CommentDTO[];
}
export enum ArticleCategory {
  Technology,
  Science,
  Sports,
  Business,
  Entertainment
}
export interface CreateArticleDTO {
  title: string;
  body: string;
  category: ArticleCategory;
}

export interface EditArticleDTO {
  id: number;
  title: string;
  body: string;
  category: ArticleCategory;
}

File: AngularNewsAPI/AngularNewsAPI/Data/Entities/Article.cs
using System.ComponentModel;
using System.Text.Json;

namespace AngularNewsAPI.Data.Entities
{
    public enum Category
    {
        Technology,
        Health,
        Sports,
        Entertainment,
        Business,
        Science,
        World
    }
    public class Article
    {
        public int Id { get; set; }
        public string Title { get; set; }
        public string Body { get; set; }
        public DateTime PublishedAt { get; set; }
        public int Views { get; set; }
        public int Rating { get; set; }
        public List<Comment> Comments { get; set; }
        public Category Category { get; set; }
        public int AuthorId { get; set; }
        public User Author { get; set; }
    }
}